### PR TITLE
Stage 1

### PR DESCRIPTION
**Pull request by vivek-ncompass**
Create user, blog and topic

login user

update blog and topic

made guards wherever required

### CODE REVIEW

File: src/auth/guard/blogCreatorVerification.guard.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest()
+    const body = request.body
+    
+    try{
+      const topicDetails = await this.topicsRepository.findOne({where: {topic_name:body.topic}, relations:["editors"]})
+
+      const editors = topicDetails.editors
+      for(let i = 0; i<editors.length; i++){
```
Comment (praveen-ncompass) [inline src/auth/guard/blogCreatorVerification.guard.ts]: optimize this if possible

File: src/users/users.service.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  }
+    
+  async changePassword(id: number, changePasswordUser : ChangePasswordType){
+    const user = await this.userRepository.findOne({where: {id}})
+    if(!user){
+      throw new NotFoundException('User Not Found')
+    }
+    user.updatedAt = new Date();
+    user.password = md5(changePasswordUser.password)
```
Comment (praveen-ncompass) [inline src/users/users.service.ts]: create a separate function

File: src/auth/dtos/UserLoginDto.dto.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { IsNotEmpty, IsString } from "class-validator";
+
+export class UserLoginDto{
+
+    @IsString() @IsNotEmpty()
+    username: string
+
+    @IsString() @IsNotEmpty() 
+    password: string
```
Comment (arjun-ncompass) [inline src/auth/dtos/UserLoginDto.dto.ts]: Have the decorators in different lines

File: src/auth/guard/blogEditorVerification.guard.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+      const topicDetails = await this.topicsRepository.findOne({where:{topic_name:blogDetails.topic}, relations:['editors']})
+
+      const editors = topicDetails.editors
+
+      for(let i = 0; i<editors.length; i++){
+        if(editors[i].username === request.payload.username){
+          return true
+        }
+      }
```
Comment (arjun-ncompass) [inline src/auth/guard/blogEditorVerification.guard.ts]: Avoid using this type of looping, use `for...of`

File: src/topic/topic.service.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    topicData.desc = updateTopicParams.desc
+
+    let newEditorsArr:Users[] = [], newViewersArr:Users[] = []
+
+    for(let i = 0; i< updateTopicParams.editors.length; i++){
+      newEditorsArr.push(await this.findUserByUsername(updateTopicParams.editors[i]))
+    }
+    newEditorsArr.length>0?topicData.editors.push(...newEditorsArr):null
```
Comment (arjun-ncompass) [inline src/topic/topic.service.ts:61]: Make this a single line if condition

File: src/users/types/changePassword.type.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export type ChangePasswordType = {
+
+    password?: string;
+
+}
```
Comment (arjun-ncompass) [inline src/users/types/changePassword.type.ts]: This seems unneccessary

**Review by arjun-ncompass — COMMENTED** [review]
I noticed multiple places for the changes said but have only mentioned it once, so make the changes in places required

Also could sort the imports part in all the files, not a compulsion to add lines of spaces in between, sorting is enough

**Pull request merged**